### DISCUSSION TITLE
Is the Amission webhooks mandatory?

### DISCUSSION THREAD

**Discussion by mcantinqc** [kedacore/charts#462] [Q&A] [answered]
Hello,

We are currently utilizing EKS with a custom CNI. In order for the Admission webhooks to function properly, the `hostNetwork: true` attribute needs to be present on the pod. You can find detailed information about this issue in the following article: https://medium.com/@denisstortisilva/kubernetes-eks-calico-and-custom-admission-webhooks-a2956b49bd0d.

Unfortunately, the current chart does not offer the option to include hostNetwork: true, and although I can submit a pull request to add it, but this can potentially create conflicts with other controllers because the webhook port is not configurable.

I noticed two pull requests from @zroubalik in the repository: https://github.com/kedacore/charts/pull/451 and https://github.com/kedacore/charts/pull/453. These pull requests state that "Admission webhooks won't be mandatory in v2.12." Therefore, I assume that they are required in version 2.10. However, the comment mentions that "this feature option will be removed in v2.12," which contradicts the title of the pull request. I am confused about this inconsistency.

To summarize, my questions are:

Are the Admission webhooks mandatory in the current version 2.10?
Will the Admission webhooks be mandatory in the future version 2.12?

**Comment by tomkerkhove** [comment, marked as answer]
Nope, you can just use it without. It helps enforcing best practices, if you want.

**Reply by zroubalik** [reply]
It is and will be an optional component. @mcantinqc would be nice if you can open an issue to track the `hostnetwork` concern and even nicer if you can contribute a fix. That sounds completely reasonable. Thanks!
